googledest/users.go

```diff
@@ -247,3 +265,177 @@ func (g *GoogleUsers) updateUser(
 
 	atomic.AddUint64(counter, 1)
 }
+
+func (g *GoogleUsers) getUser(email string) (admin.User, error) {
+	userCall := g.AdminService.Users.Get(email)
+	user, err := userCall.Do()
+	if err != nil || user == nil {
+		return admin.User{}, err
+	}
+	return *user, nil
+}
+
+func updateIDs(newID string, oldIDs interface{}) ([]admin.UserExternalId, error) {
+	IDs := []admin.UserExternalId{{
+		Type:  "organization",
+		Value: newID,
+	}}
+
+	if oldIDs == nil {
+		return IDs, nil
+	}
+
+	interfaces, ok := oldIDs.([]interface{})
+	if !ok {
+		return nil, errors.New("unexpected data in Google API ID list")
+	}
+
+	for i := range interfaces {
+		IDMap, ok := interfaces[i].(map[string]interface{})
+		if !ok {
+			return nil, errors.New("unexpected data in Google API ID list")
```

Comment:
Should this error message be at least slightly different than the others to help with troubleshooting?

Reply:
Good idea.